client/client_test.go

```diff
@@ -42,6 +45,15 @@ func TestClient_Cred_PathMissing(t *testing.T) {
 	}
 }
 
+func TestClient_Cred_EnvOverride(t *testing.T) {
+	configFilePath := "/testpath"
+	os.Setenv("GOOGLE_API_CERTIFICATE_CONFIG", util.GetDefaultConfigFilePath())
```

Comment:
This is a good test case for a missing config, but in this case the test should check the priority of the ENV vs explicit function path.

Reply:
Liam 2 things:
1. I would not set "GOOGLE_API_CERTIFICATE_CONFIG" to util.GetDefaultConfigFilePath() for your test cases, because it becomes indistinguishable from the default behavior - instead point it to a unique value based on your test needs.
2. I would create 2 separate unit tests to test the following behaviors:
 a) If configFilePath is "" and env var is set, I should load the config from env var
 b) if configFilePath is set to "testdata/certificate_config.json" and env var is set to something else (such as "missing.json"), then I should give priority to the explict configFilePath.

Reply:
Okay, I will try that out. Thank you for the pointers!